### PR TITLE
ext: in setting editor

### PR DESCRIPTION
**Pull request by rebornix**
This PR adds support for filtering by extension id in settings editor

Single extension id
```
@ext:coenraads.bracket-pair-colorizer
```

OR operator
```
@ext:coenraads.bracket-pair-colorizer,github.vscode-pull-request-github
```

### CODE REVIEW

File: src/vs/workbench/contrib/preferences/electron-browser/settingsEditor2.ts [hunk 1]
```diff
@@ -1077,6 +1080,7 @@ export class SettingsEditor2 extends BaseEditor {
 			}
 
 			this.refreshTOCTree();
+			this.tocTree.expandAll();
```
Comment (rebornix) [inline src/vs/workbench/contrib/preferences/electron-browser/settingsEditor2.ts]: The reason that `this.tocTree.expandAll()` is added here is the first time the settings editor being opened with predefined query, the TOC is collapsed even though the tree item `collapsed` state is `undefined`.

File: src/vs/workbench/contrib/preferences/electron-browser/settingsEditor2.ts [hunk 2]
```diff
@@ -69,7 +70,7 @@ export class SettingsEditor2 extends BaseEditor {
 	private static SETTING_UPDATE_SLOW_DEBOUNCE: number = 1000;
 
 	private static readonly SUGGESTIONS: string[] = [
-		`@${MODIFIED_SETTING_TAG}`, '@tag:usesOnlineServices'
+		`@${MODIFIED_SETTING_TAG}`, '@tag:usesOnlineServices', `@${EXTENSION_SETTING_TAG}`
```
Comment (rebornix) [inline src/vs/workbench/contrib/preferences/electron-browser/settingsEditor2.ts]: If we don't want users to type in extension filters in the input, then we should not add it here.

File: src/vs/workbench/contrib/preferences/electron-browser/settingsEditor2.ts (hunk 1 above)
Comment (roblourens) [reply src/vs/workbench/contrib/preferences/electron-browser/settingsEditor2.ts]: Won't this be covered by the expandAll call above on 1071?

Comment (rebornix) [reply src/vs/workbench/contrib/preferences/electron-browser/settingsEditor2.ts]: Removed.

**Review by roblourens — APPROVED** [review]

**Pull request merged**